protocol/bc/spend.go

```diff
@@ -3,7 +3,7 @@ package bc
 // SpendInput satisfies the TypedInput interface and represents a spend transaction.
 type SpendInput struct {
 	// Commitment
-	Outpoint
+	SpentOutputID OutputID
```

Comment:
Good call about the stutter of `SpendInput.SpentOutputID` vs `SpendInput.OutputID`. But
it's best to avoid embedding unless there's a specific reason to make field and method names available on this type directly. If we want the name to be OutputID, declaring it as `OutputID OutputID` is ok.